core/providers/replicate/images.go

```
@@ -181,6 +183,166 @@ func ToBifrostImageGenerationResponse(
+}
+
+// setUpscaleOutputSize writes a resolved pixel count onto a response as a size
+// string, leaving a size the provider itself reported untouched.
+func setUpscaleOutputSize(pixels int, response *schemas.BifrostImageGenerationResponse) {
+	if pixels <= 0 {
+		return
+	}
+	if response.ImageGenerationResponseParameters == nil {
+		response.ImageGenerationResponseParameters = &schemas.ImageGenerationResponseParameters{}
+	}
+	if response.ImageGenerationResponseParameters.Size == "" {
```

golangci-lint (2.12.2)

[error] QF1008: could remove embedded field "ImageGenerationResponseParameters" from selector

(staticcheck)